python/pyarrow/array.pxi

```diff
@@ -159,9 +159,10 @@ def array(object obj, type=None, mask=None, size=None, from_pandas=None,
 
     Notes
     -----
-    Localized timestamps will currently be returned as UTC (pandas's native
-    representation). Timezone-naive data will be implicitly interpreted as
-    UTC.
+    Timezone will be preserved in the returned array for timezone-aware data,
+    else no timezone will be returned for naive timestamps.
+    Internally, UTC values are stored for timezone-aware data,
+    whereas timezone-naive data is implicitly interpreted as if in UTC.
```

Comment:
```suggestion
    Internally, UTC values are stored for timezone-aware data with the
    timezone set in the data type.
```

I would personally leave out the "as if in UTC for timezone-naive data" part, as this is a detail for how this data is interpreted internally in other kernels, but can be confusing for the general reader I think (in the end, what matters for the user is that they get a timestamp array without any timezone set).